### ISSUE TITLE
String::String(const char*) is expecting latin1 instead of UTF-8

### ISSUE THREAD

**Issue by saki7** [godotengine/godot-cpp#784]
Since #695, `String::String(const char*)` is now internally calling `string_new_with_latin1_chars` instead of `string_new_with_utf8_chars`.

diff: https://github.com/godotengine/godot-cpp/commit/bf8fc4c53d07cf8ac4d8343364899c197076fbc2#diff-20a0dfd8072691f654c12adb6aecabee9d3247fb99488b37539dab1d967c97b8L77-L79

Because of this change, we can't pass non-latin1 characters for String-like parameters anymore.
For example, `node->get_node("Path/To/NonLatin1/名前")` fails.

Technically speaking, `char*` is encoded in a native encoding, which is not guaranteed to be UTF-8 (by definition). Thus, we should not choose between `latin1` and `utf8` in the first place; we should add a new function `string_new_with_native_chars` and call it whenever `char*` is used as an input. But that's a very long story. For sake of simplicity, we can just expect UTF-8 for `char*`, because it is the default for many environments today (note that MSVC can be configured to use UTF-8 by the `/utf-8` compiler flag).

No offense, but even if the change of the default encoding was intended, it should not be done in #695 because it is a breaking change. I consider this a regression.

**Comment by Calinou** [comment]
cc @bruvzg

**Comment by bruvzg** [comment]
> Since https://github.com/godotengine/godot-cpp/pull/695, String::String(const char*) is now internally calling string_new_with_latin1_chars instead of string_new_with_utf8_chars.

This is intentional, it's the same behavior as the Godot `String` is using in the engine code. Since the same code can be used as both a built-in module and extension, the behavior should be identical.

> a breaking change. I consider this a regression.

GDExtension API is not finalized and pretty much every change is breaking.

**Comment by bruvzg** [comment]
I agree that using UTF-8 is better, but this change should be done in both engine and gdextension at the same time (which will involve a lot of changes to the engine).

**Comment by saki7** [comment]
Should we ping other committers? Since I'm not familiar with the organization I'm not sure how we could reach a consensus.

**Comment by bruvzg** [comment]
> Should we ping other committers? Since I'm not familiar with the organization I'm not sure how we could reach a consensus.

I do not think it's something big enough to require a formal proposal (I'm pretty sure it was proposed multiple times before, and no one opposed it, but no one bother to implement it either).

Anyway, I have already added it to my TODO list.
